src/main/java/org/opensearch/security/support/Base64Helper.java

```diff
@@ -220,7 +229,11 @@ private SafeObjectOutputStream(OutputStream out) throws IOException {
 
         @Override
         protected void writeClassDescriptor(final ObjectStreamClass desc) throws IOException {
-            super.writeClassDescriptor(descriptorReplacer.replace(desc));
+            if (replaceWithOdfePackage) {
+                super.writeClassDescriptor(descriptorReplacer.replace(desc));
+            } else {
+                super.writeClassDescriptor(desc);
+            }
         }
```

Comment:
For my knowledge, what role does this method play here?

Reply:
This is where the action happens during serialization. To make sure rolling upgrades worked between ODFE-1.13 and OS 1 this was added to the code to rewrite `org.opensearch` with `com.amazon.opendistroforelasticsearch` to that when the ODFE node received a transport action it would be able to understand what was contained in the action. ODFE nodes have no knowledge of the `org.opensearch` package name.

This conditionally adds the `descriptorReplacer` if the cluster has ODFE nodes.

Reply:
Ahh..got it..thank you!